src/19FAF0.c

```diff
@@ -223,7 +225,739 @@ s32 calc_player_test_enemy(void) {
     return 0;
 }
 
-INCLUDE_ASM(s32, "19FAF0", calc_player_damage_enemy);
+s32 calc_player_damage_enemy(void) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* player = battleStatus->playerActor;
+    s32 currentTargetID = battleStatus->currentTargetID;
+    s32 currentTargetPartID = battleStatus->currentTargetPart;
+    ActorState* state;
+    Evt* evt;
+    Actor* target;
+    ActorPart* targetPart;
+    s32 retVal;
+    s32 currentAttackDamage;
+    s32 targetDefense;
+    s32 dispatchEvent;
+    s32 sp1C;
+    s32 sp20;
+    s32 sp24;
+    s32 isFireDamage;
+    s32 isElectricDamage;
+    s32 isIceDamage;
+    s32 tempBinary;
+    s32 wasStatusInflicted;
+    s32 isAbilityActive;
+
+    sp20 = FALSE;
+    isFireDamage = FALSE;
+    isElectricDamage = FALSE;
+    isIceDamage = FALSE;
+    tempBinary = FALSE;
+    sp1C = FALSE;
+    wasStatusInflicted = FALSE;
+    sp24 = FALSE;
```

Comment:
These can be merged with their declarations above

Reply:
I tried that but it causes issues with the order that stack pointers are declared in the assembly